src/sidebar/util/term-before-position.ts

```diff
@@ -28,16 +26,20 @@ export function getContainingWordOffsets(
   text: string,
   referencePosition: number,
 ): WordOffsets {
-  const precedingEmptyCharPos = text
-    .slice(0, referencePosition)
-    .search(/\s\S*$/);
-  const subsequentEmptyCharPos = text.slice(referencePosition).search(/\s/);
+  const precedingText = text.slice(0, referencePosition);
+  const matches = [...precedingText.matchAll(/[\s,.;:|?!'"\-()[\]{}]/g)];
```

Comment:
If you want to share subcomponents of a regex you can always construct a regex using a non-literal (ie. `new RegEx(pattern)` where `pattern` is a string that can use interpolation etc.)

Reply:
That's the thing. I'm not completely sure that's going to make things more clear, or less, if I see something like ``new RegExp(`/[${SEPERATOR_CHARACTERS}]/g`)`` or ``new RegExp(`/(^|[${SEPERATOR_CHARACTERS}])@([A-Za-z0-9._]+)(?=[${SEPERATOR_CHARACTERS}]|$)/g`)`` assuming `const SEPERATOR_CHARACTERS = '\s,.;:|?!'"\-()[\]{}'`.

What do you think?

Reply:
I have decided to not do this finally, because I see some functions may not want to match all the punctuation characters, which would make this generalization even more complex to handle.

What I have done is moving all the helpers to the `mentions.ts` module, so at least all the affected regular expressions are now in a single place and allow us to more easily revisit any decisions in future.